### PR TITLE
fix if order which defines output fc layer

### PR DESCRIPTION
**Pull request by sungchul1**
This PR contains a bugfix about the model with AngleSimpleLinear is not loaded even though self.loss is set to 'am_softmax'.

### CODE REVIEW

File: torchreid/models/efficient_net_pytcv.py [hunk 1]
```diff
@@ -351,13 +351,13 @@ def __init__(self,
...
+            self.output.add_module("asl", AngleSimpleLinear(
                 in_features=final_block_channels,
                 out_features=self.num_classes))
         else:
-            assert 'am_softmax' in self.loss or 'am_binary' in self.loss
-            self.output.add_module("asl", AngleSimpleLinear(
+            assert 'softmax' in self.loss or 'asl' in self.loss
+            self.output.add_module("fc", nn.Linear(
                 in_features=final_block_channels,
                 out_features=self.num_classes))
```
Comment (vinnamkim) [inline torchreid/models/efficient_net_pytcv.py]: How about
```
if ~ or ~:
    self.output.add_module(~)
elif ~ or ~:
    self.output.add_module(~)
else:
    raise ValueError(f"self.loss={self.loss} is unknown loss type.")
```

Comment (sungchul1) [reply torchreid/models/efficient_net_pytcv.py]: Done.

**Review by vinnamkim — APPROVED** [review]
LGTM.
